Update onboarding task documentation & add back `example` command

chihsuan opened the pull request:
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

<!-- If necessary, indicate if this PR is part of a bigger feature. Add a label with the format `focus: name of the feature [team:name of the team]`. -->

<!-- Describe the changes made to this Pull Request and the reason for such changes. -->

Related https://github.com/woocommerce/woocommerce/issues/33293

Update onboarding task documentation 

1. Add back `pnpm --filter=@woocommerce/admin-library example` command so that we can build the examples.
2. Update the example's webpack config to ignore the output directory to avoid infinite rebuilds
3. Improve onboarding task documentation
     - Add readme to add task example
     - Update `docs/extension-development/handling-merchant-onboarding.md` (It's outdated.)

<!-- Begin testing instructions -->

### How to test the changes in this Pull Request:

<!-- Include detailed instructions on how these changes can be tested. Review and follow the guide for how to write high-quality testing instructions. -->

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Review the documentation changes.
2. Run `WC_EXT=add-task pnpm --filter=@woocommerce/admin-library example`
3. Confirm `plugins/add-task` is created.
4. Activate the plugin and confirm the task is added to the list.
5. Run `WC_EXT=add-task pnpm --filter=@woocommerce/admin-library example --watch`
6.  Confirm it watches for changes and only rebuilds when the file changes.

<!-- End testing instructions -->

### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box and supplying data. -->

-   [x] Automatically create a changelog entry from the details below.

<details>

#### Significance

<!-- Choose only one -->

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [x] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message <!-- Add a changelog message here -->

Update onboarding task documentation and fix example command

#### Comment <!-- If the changes in this pull request don't warrant a changelog entry, you can alternatively supply a comment here. Note that comments are only accepted with a significance of "Patch" -->

</details>

chihsuan committed:
Add back example command to wca package.json

chihsuan committed:
Ignore the output directory to avoid an infinite loop.

chihsuan committed:
Improve onboarding task docs

Commit:
Add changefile(s) from automation for the following project(s): woocommerce

chihsuan committed:
Fix lint

psealock commented on plugins/woocommerce-admin/docs/examples/extensions/examples.config.js:
```diff
@@ -103,6 +103,12 @@ const webpackConfig = {
 	],
+	watchOptions: {
+		ignored: [
+			path.resolve( __dirname, `../../../../${ extension }/` ),
```
When I change any file, say `plugins/add-task/js/index.js` it doesn't rebuild. Could this `ignore` be causing that?

chihsuan replied:
Nope, this is unrelated but I've just changed it to ignore `dist` folder only.

psealock reviewed (commented):
Hi @chihsuan, this is working well. I wasn't able to get the example plugin to rebuild though.

chihsuan committed:
Change output path and update ignored option

Change output path to plugin folder directly because copy-webpack-plugin is not designed to copy files generated from the build process;

chihsuan committed:
Update docs/docs-manifest.json

chihsuan commented:
> Hi @chihsuan, this is working well. I wasn't able to get the example plugin to rebuild though.

Hey @psealock Good catch! I found that it requires saving twice to refresh because [`copy-webpack-plugin`](https://webpack.js.org/plugins/copy-webpack-plugin/) is **not designed to copy files generated from the build process**.

I changed the output path to the path under the plugin folder in b3ac4253fd518899690f66f395a6945174e58f8e to fix it. Could you please try again? You might need to remove `plugins/woocommerce-admin/docs/examples/extensions/add-task/dist` folder first before testing. 🙏

ilyasfoo commented on plugins/woocommerce-admin/docs/examples/extensions/add-task/README.md:
````diff
@@ -0,0 +1,24 @@
+
+```bash
+WC_EXT=add-task pnpm --filter=@woocommerce/admin-library example --watch
+```
````
When I initially tried this, I modified the built file in `plugins/add-task/js/index.js`, which did nothing. Can we also add the detail on where to make the changes? I think it's an important detail to add. What do you think?

chihsuan replied:
Sounds good! Added more details in ca5a1ba6d7b40cf371f0d8167ee9bcaf3cee0652

ilyasfoo reviewed (approved):
Thanks, @chihsuan! Tested well, but I have a minor comment. Pre-approving

chihsuan committed:
Update add-task/README.md

psealock commented:
Thanks for the changes, when I run `WC_EXT=add-task pnpm --filter=@woocommerce/admin-library example` I get the following error, look familiar?

```
ERROR in Conflict: Multiple assets emit different content to the same filename index.js.map
```

chihsuan commented:
> Thanks for the changes, when I run `WC_EXT=add-task pnpm --filter=@woocommerce/admin-library example` I get the following error, look familiar?
> 
> ```
> ERROR in Conflict: Multiple assets emit different content to the same filename index.js.map
> ```

@psealock Yes, did you remove the `plugins/woocommerce-admin/docs/examples/extensions/add-task/dist` folder? 🤔

psealock reviewed (approved):
Thanks for the clarification @chihsuan, this is working well now 🚢
